test/e2e_node/services/kubelet.go

```diff
@@ -190,6 +197,58 @@ func createPodManifestDirectory() (string, error) {
 	return path, nil
 }
 
+// createKubeconfig creates a kubeconfig file at the fully qualified `path`.
+func createKubeconfig(path string) error {
+	kubeconfig := `apiVersion: v1
+kind: Config
+users:
+- name: kubelet
+clusters:
+- cluster:
+    server: ` + getAPIServerClientURL() + `
+    insecure-skip-tls-verify: true
```

Comment:
Even if all the tests run with secured localhost, that just validates that secured localhost is available. We'd need a test that explicitly tries to use unsecured localhost to ensure it is actually locked down.